### ISSUE TITLE
Описание протокола CoAP (RFC7252)

### ISSUE THREAD

**Issue by uralbash** [ustu/lectures.www#47]
https://tools.ietf.org/html/rfc7252

**Comment by AlenaPankova** [comment]
+

**Comment by AlenaPankova** [comment]
Здравствуйте. Высылаю примерный текст доклада.
[CoAP.docx](https://github.com/ustu/lectures.www/files/1407884/CoAP.docx)

**Comment by uralbash** [comment]
@AlenaPankova 

Необходимо раскрыть в докладе следующие темы:

* Из чего состоит URI
* Какой порт по умолчанию
* Методы и их коды
* Ответы и их коды
* Заголовки и их коды
* Типы контента
* Описать уровни протокола
```
                         +----------------------+
                         |      Application     |
                         +----------------------+
                         +----------------------+
                         |  Requests/Responses  |
                         |----------------------|  CoAP
                         |       Messages       |
                         +----------------------+
                         +----------------------+
                         |         DTLS         |
                         +----------------------+
                         +----------------------+
                         |          UDP         |
                         +----------------------+
```
* Описать как при помощи `Сonfirmable` происходит подтверждение доставки поUDP
* Привести пример, можно использовать https://github.com/Tanganelli/CoAPthon и вместо клиента https://addons.mozilla.org/ru/firefox/addon/copper-270430/
* Отличия от протокола `HTTP`

**Comment by AlenaPankova** [comment]
Дополнила: [CoAP.docx](https://github.com/ustu/lectures.www/files/1412277/CoAP.docx)
У меня есть пара вопросов: Вопрос про коды методов и заголовков, по сути ведь нужно было расписать коды ответов или нужно что-то еще? И что нужно показать в примере?

**Comment by uralbash** [comment]
@AlenaPankova 
Да, просто описание кодов. В примере нужно поднять сервер, отправить любой запрос и получить ответ.

**Comment by AlenaPankova** [comment]
Презентация: [CoAP.pptx](https://github.com/ustu/lectures.www/files/1414562/CoAP.pptx)
Текстовка: [CoAP.docx](https://github.com/ustu/lectures.www/files/1414563/CoAP.docx)
Пойдет?

**Comment by uralbash** [comment]
@AlenaPankova да сойдет

**Comment by theedanico** [comment]
Взял данную тему.
